calblueprint/siren-web
# Andrei/intake form page

# ✨ New in this PR ✨

## One liner: what did you do? 🚨 
Integrated question component and intake form overlay, can add to question list and reorder it by dragging.

## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [ x] Functioning add button.
 - [ x] Display and generate list of questions
 - [ x] Reorder list using drag-and-drop.  Establish persistence with question information.

## How can the reviewer test your code? 👩‍💻
Go into intake form page.  Click add circle to create generic questions.  Add information into questions.  Drag and reorder list, make sure they go into the right spots.

## Any bugs you encountered or still having trouble with? 🐛

EDIT: resolved 
Pending Bug:   onDragEnd is really inconsistent.  Draggable objects that are in the questions array are assigned indices on creation.  onDragEnd should be changing these indices as it gets reordered, but sometimes the program doesn't do that (not sure how to isolate this fully).  i.e a draggable created at index 0 is moved to index 3, but it is still referred to as index 0 in source.index which is used in onDragEnd.  Causes "jumps" when reordering list. 
Main issue with splice.  To replicate: 
1. in onDragEnd, add console.log(source.index) and console.log(destination.index)
2. Can also add 0, 1, 2 , ... in order to question components in-browser to track original placement of components
3. Console.log("some message") to the 2nd if statement in onDragEnd, reordering doesn't happen if index of source and destination match.
4. I'd mess around with first and last indices the most, most extreme errors. _


## Resources 📔
_link online resources and related PRs_ 


🧜‍♀️ cc: @shannonbonet @phoebeli23
